scarlet/update.py

```diff
@@ -214,13 +216,106 @@ def translation(component, direction=1, kernel=interpolation.lanczos, padding=3)
     return component
 
 
-def symmetric(component, strength=1, use_prox=True, kernel=interpolation.lanczos, padding=3):
+def psf_weighted_centroid(component):
+    # Extract the arrays from the component
+    morph = component.morph
+    if component.frame.psfs is None:
+        if not hasattr(component, "_centroid_psf"):
```

Comment:
Hard coded shape??
I don't see why we should create a psf if the user did not specify one.

Reply:
This is really just Jim's function to weight the center of flux to give more weight to the central pixels. So in reality any function that falls off from the peak can be used, so think of `centroid_psf` as `centroid_weight`. Perhaps I should just rename it to make its use more clear.

Reply:
I am in favour of renaming. If I read this correctly, this `weight map` only serves to get the centroids?
If so, I am good with that.

Reply:
Although, could it happen that `component.pixel_center` be way off? (e.g heavily blended scene)
In this case I am worried that this weigthing might bias the centering.

Reply:
While I can't say for sure, I don't think so. This isn't a traditional centroid, in that the centroid is acting on the *model*, not the *image*. So while the image would see interference in the center of flux measurement from neighbors, after the first few iterations the model center of flux should be more stable. Also, `pixel_center` is the point about which we are making the solution monotonic so if it is way off then our entire model will be off.